nemo/lightning/megatron_parallel.py

```diff
@@ -587,7 +587,7 @@ def init_model_parallel(self):
         from megatron.core.tensor_parallel.layers import set_defaults_if_not_set_tensor_model_parallel_attributes
 
         for model_module in self:
-            if not self._cpu and (not HAVE_CUSTOM_FSDP or not self.ddp_config.use_custom_fsdp):
+            if not self._cpu and not (HAVE_CUSTOM_FSDP and self.ddp_config and self.ddp_config.use_custom_fsdp):
```

Comment:
I had to add this locally where I am using NeMo from my branch and Mcore ToT and hitting `self.ddp_config.use_custom_fsdp` but since for pruning we don't use DDP, the config is None. But in CI this line was not hit (maybe `HAVE_CUSTOM_FSDP=False`).

Regardless I think its better to check `if self.ddp_config is not None` before checking its attribute hence added the change